### ISSUE TITLE
Fix VB Parsing of Format Specifiers

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#58043]
VB has the same parsing "bug" as detailed in https://github.com/dotnet/roslyn/issues/57750, but unlike C# it does not support interpolated string handlers, so there are no runtime impacts to the parsing issue. We should update the VB parser to be in sync with the C# parser and the runtime here, but it's not high priority so the team is not going to focus on it. We'd accept a community PR to implement this fix.

**Comment by CyrusNajmabadi** [comment]
Impl of the fix should likely be similar to: https://github.com/dotnet/roslyn/pull/57751

**Comment by 333fred** [comment]
There are no plans to add interpolated string handlers to VB. We'd likely accept a refactoring, if submitted by the community.

**Comment by CyrusNajmabadi** [comment]
> The refactoring will be a dangerous solution since there is an intent to prevent the <Obsolete> trick to allow vb to use ref records.

I have no clue what this is referring to.

**Comment by CyrusNajmabadi** [comment]
> Or, at least provide a refactoring to convert interpolated strings to a method call that uses interpolated handlers

This is something you can write yourself.  I don't see this being something we would particularly add here as the point is to be concise.  Once you're being this verbose, might as well write it by hand.

**Comment by KathleenDollard** [comment]
We do not plan to add interpolated strings to VB. Our decision here is consistent with our language strategy.

I am not sure of the intent of the phrase "you better" above. In common US English usage that is considered a threat. We do not tolerate that. Please pick different words in the future.

**Comment by VBAndCs** [comment]
@KathleenDollard 
Expressions in any language take there meaning from context, so, they can mean different things. According to [this dictionary](https://www.urbandictionary.com/define.php?term=you+better) it is used also for advice:  
> you better 
> Same as "you should", though it can be use for any subject (I, he, she, we, it, they). Implies advice or warning. Common in the southern US, this phrase has been shortened down from:

You would [be better](https://www.urbandictionary.com/define.php?term=be%20better) off if you...
You would be better to ...
You'd be better...
You'd better...
[You better](https://www.urbandictionary.com/define.php?term=You%20better)...

**Comment by VBAndCs** [comment]
@KathleenDollard 
> We do not plan to add interpolated strings to VB. Our decision here is consistent with our language strategy.

I think this improvement doesn't break this strategy. It just gives it better performance:
> Going forward, we do not plan to evolve Visual Basic as a language. This supports language stability and maintains compatibility between the .NET Core and .NET Framework versions of Visual Basic.

U think things should be clear after MS created Razor Pages, Blazor m WinUI 3, and MAUI:
VB.NET developers were absolutely right to stuck with WinForms and WebForms (some with WPF and SilverLight) and refuse to keep up with MS frameworks sudden changes. VB is still holding up its position right behind C# in popularity index although it is frozen as a language since 2017. I think it is time to review this strategy. VB.NET has no issue with the new techs. They just were enforced badly on the business market and MS couldn't make devs follow up its dreams.

**Comment by VBAndCs** [comment]
@CyrusNajmabadi 
> I have no clue what this is referring to.

https://github.com/dotnet/roslyn/issues/50118

**Comment by CyrusNajmabadi** [comment]
@VBAndCs Yes.  taht will likely break at some point.  Please don't write any code that depends on it.  It is supported and will almost certainly not work with future work coming down the line.  As per our conversations in other venues, depending on this will be a problem.

**Comment by CyrusNajmabadi** [comment]
@VBAndCs from your own line: "Implies advice or warning."

The "implies warning" part is something you may not have been aware of.  We understand that that might not be clear being ESL, so just keep this in mind for the future to help prevent these sorts of miscommunications.  Thanks! :)

**Comment by CyrusNajmabadi** [comment]
@VBAndCs I've hidden yoru post as being off topic.  This post is about parsing VB format specifiers.  It's not a general place for you to air grievances you have with our strategy, which we've been very clear on for over 5 years at this point.

**Comment by VBAndCs** [comment]
@CyrusNajmabadi 
> Please don't write any code that depends on it. It is not supported and will almost certainly not work with future work coming down the line. As per our conversations in other venues, depending on this will be a problem.

I will be able to workaround this by setting an old `LangVersion`, and since VB not expected to have new features, this will not be a big deal, and if it is, I can separate such functionality into a dll. As I always say: VB devs are the workaround masters :).
Thanks.

**Comment by CyrusNajmabadi** [comment]
@VBAndCs I've hidden yoru post as being off topic. This post is about parsing VB format specifiers. It's not a general place for you to air grievances you have with our strategy, which we've been very clear on for over 5 years at this point.
